unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Materials/Data/MaterialModel.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Helpers;
+using UnityEngine;
+
+[System.Serializable]
+public class MaterialModel : BaseModel
+{
+    public enum TransparencyMode
+    {
+        OPAQUE,
+        ALPHA_TEST,
+        ALPHA_BLEND,
+        ALPHA_TEST_AND_BLEND,
+        AUTO
+    }
+    
+    [Range(0f, 1f)]
+    public float alphaTest = 0.5f;
+
+    public Color albedoColor = Color.white;
+
+    public float metallic = 0.5f;
+    public float roughness = 0.5f;
+    public float microSurface = 1f; // Glossiness
+    public float specularIntensity = 1f;
+    
+    public Color emissiveColor = Color.black;
+    public float emissiveIntensity = 2f;
+    public Color reflectivityColor = Color.white;
+    public float directIntensity = 1f;
+
+    public bool castShadows = true;
+    
+    [Range(0, 4)]
+    public int transparencyMode = 4; // 0: OPAQUE; 1: ALPHATEST; 2: ALPHBLEND; 3: ALPHATESTANDBLEND; 4: AUTO (Engine decide)
+    
+    public TextureModel albedoTexture;
+    public TextureModel alphaTexture;
+    public TextureModel emissiveTexture;
+    public TextureModel bumpTexture;
+
+    public override BaseModel GetDataFromJSON(string json) { return Utils.SafeFromJson<MaterialModel>(json); }
+    
+        public override int GetHashCode()
+    {
+        unchecked
+        {
+            int hashCode = alphaTest.GetHashCode();
+            hashCode = (hashCode * 397) ^ albedoColor.GetHashCode();
+            hashCode = (hashCode * 397) ^ metallic.GetHashCode();
+            hashCode = (hashCode * 397) ^ roughness.GetHashCode();
+            hashCode = (hashCode * 397) ^ microSurface.GetHashCode();
+            hashCode = (hashCode * 397) ^ specularIntensity.GetHashCode();
+            hashCode = (hashCode * 397) ^ emissiveColor.GetHashCode();
+            hashCode = (hashCode * 397) ^ emissiveIntensity.GetHashCode();
+            hashCode = (hashCode * 397) ^ reflectivityColor.GetHashCode();
+            hashCode = (hashCode * 397) ^ directIntensity.GetHashCode();
+            hashCode = (hashCode * 397) ^ castShadows.GetHashCode();
+            hashCode = (hashCode * 397) ^ transparencyMode;
+            hashCode = (hashCode * 397) ^ (albedoTexture != null ? albedoTexture.GetHashCode() : 0);
+            hashCode = (hashCode * 397) ^ (alphaTexture != null ? alphaTexture.GetHashCode() : 0);
+            hashCode = (hashCode * 397) ^ (emissiveTexture != null ? emissiveTexture.GetHashCode() : 0);
+            hashCode = (hashCode * 397) ^ (bumpTexture != null ? bumpTexture.GetHashCode() : 0);
+            return hashCode;
+        }
+    }
```

Comment:
if we need to add this, I would call it with another name, without overriding `GetHashCode`
 `GetHashCode` should not depend on field values and should be immutable for the whole life of an instance.
 overriding it could generate a lot of issues that are not easy no find or debug.
 ie:
```
 Obj obj = new Obj()
 HashSet hashSet  = add(obj)
 obj.someField = change
 hashSet.contains(obj) == false
```

Reply:
We need this override since the Dictionaries uses it under the hood to work, We can't escape from overridering it. 

In order to escape, we should create our own implementation of Dictionary and use it in all the asset promises keeper.

The only thing that we can do is generate the hashCode at the creation of the object